ios/RN/RNCamera.m

```diff
@@ -423,33 +414,44 @@ - (void)record:(NSDictionary *)options resolve:(RCTPromiseResolveBlock)resolve r
 
     [self updateSessionAudioIsMuted:!!options[@"mute"]];
 
-    AVCaptureConnection *connection = [self.movieFileOutput connectionWithMediaType:AVMediaTypeVideo];
-    [connection setVideoOrientation:[RNCameraUtils videoOrientationForInterfaceOrientation:[[UIApplication sharedApplication] statusBarOrientation]]];
-
-    if (options[@"codec"]) {
-      AVVideoCodecType videoCodecType = options[@"codec"];
-      if (@available(iOS 10, *)) {
-        if ([self.movieFileOutput.availableVideoCodecTypes containsObject:videoCodecType]) {
-          [self.movieFileOutput setOutputSettings:@{AVVideoCodecKey:videoCodecType} forConnection:connection];
-          self.videoCodecType = videoCodecType;
-        } else {
-          RCTLogWarn(@"%s: Video Codec '%@' is not supported on this device.", __func__, videoCodecType);
-        }
-      } else {
-        RCTLogWarn(@"%s: Setting videoCodec is only supported above iOS version 10.", __func__);
-      }
-    }
-
     dispatch_async(self.sessionQueue, ^{
         [self updateFlashMode];
-        NSString *path = [RNFileSystem generatePathInDirectory:[[RNFileSystem cacheDirectoryPath] stringByAppendingString:@"Camera"] withExtension:@".mov"];
-        NSURL *outputURL = [[NSURL alloc] initFileURLWithPath:path];
-        [self.movieFileOutput startRecordingToOutputFileURL:outputURL recordingDelegate:self];
+
+        if (self.videoWriter.status != AVAssetWriterStatusWriting) {
+            [self.videoWriter startWriting];
+            [self.videoWriter startSessionAtSourceTime:self.bufferTimestamp];
+            self.canAppendBuffer = YES;
+            dispatch_async(dispatch_get_main_queue(), ^{
+                if (options[@"maxDuration"]) {
+                    Float64 maxDuration = [options[@"maxDuration"] floatValue];
+                    self.timer = [NSTimer scheduledTimerWithTimeInterval:maxDuration target:self selector:@selector(stopAssetWriter) userInfo:nil repeats:NO];
+                    return;
+                }
+                self.timer = [NSTimer scheduledTimerWithTimeInterval:3.0 target:self selector:@selector(stopAssetWriter) userInfo:nil repeats:NO];
```

Comment:
Rather than scheduling an arbitrary timer length if there is no `maxDuration` can we instead bridge the `stop` action and just require that to be called manually?